src/eddymotion/model/dipy.py

```diff
@@ -330,6 +332,55 @@ def compute_spherical_covariance(
     return np.where(theta <= a, 1 - 3 * (theta / a) ** 2 + 2 * (theta / a) ** 3, 0)
 
 
+def compute_derivative(
+    theta: np.ndarray,
+    kernel: np.ndarray,
+    weighting: str,
+    params: dict[float],
+):
+    """
+    Compute the analytical derivative of the kernel.
+
+    Parameters
+    ----------
+    theta : :obj:`~numpy.ndarray`
+        Pairwise angles across diffusion gradient encoding directions.
+    kernel : :obj:`~numpy.ndarray`
+        Current kernel.
+    weighting : :obj:`str`
+        The kind of kernel which derivatives will be calculated.
+    params : :obj:`dict`
+        Current parameter values.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Gradients of the kernel.
+
+    """
+
+    n_partials = len(params)
+
+    a = params.pop("a")
+    lambda_s = params.pop("lambda_s")
+
+    min_angles = theta > a
+
+    if weighting == "spherical":
+        deriv_a = (3 * theta[min_angles] / a**2) - (1.5 * (theta[min_angles] / a) ** 2) / a
```

Comment:
Missing here a 2 in the denominator of the first term (2 * a**2) ?

Reply:
Thanks for forcing me to look into this :D - I believe:

$$
\frac{\partial}{\partial a}\left( 1 - \frac{3 \theta}{2 a} + \frac{\theta^3}{2 a^3} \right) = - 1.5 \cdot \theta \frac{\partial a^{-1}}{\partial a} + \frac{\theta^3}{2} \frac{\partial a^{-3}}{\partial a} = \frac{3}{2}\left( \frac{\theta}{a^2} - \frac{\theta^3}{a^4} \right)
$$

I've revised the kernel function, which I believe was wrong (sorry for the oversight when I reviewed your PR) and now this one:

```suggestion
        deriv_a = 1.5 * (theta[min_angles] / a ** 2 - theta[min_angles] ** 3 / a ** 4)
```